src/main/java/com/videoclub/suriken/service/impl/MovieServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.videoclub.suriken.service.impl;
+
+import com.videoclub.suriken.model.Movie;
+import com.videoclub.suriken.model.MovieRenter;
+import com.videoclub.suriken.repository.MovieRepository;
+import com.videoclub.suriken.repository.RenterRepository;
+import com.videoclub.suriken.service.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+    private final MovieRepository movieRepository;
+    private final RenterRepository renterRepository;
+
+    @Autowired
+    public MovieServiceImpl(MovieRepository movieRepository, RenterRepository renterRepository) {
+        this.movieRepository = movieRepository;
+        this.renterRepository = renterRepository;
+    }
+
+
+    @Override
+    public void addMovieToDb(Movie movie) {
+        movie.setMovGenre(Movie.Genre.Drama);
+        movieRepository.save(movie);
+    }
+
+    @Override
+    public List<Movie> getAllMovies() {
+        List<Movie> movies = movieRepository.findAll();
+
+        movies.forEach(movie -> System.out.println("Movie" + movie.getMovName() + "has :" + movie.getMovieRenters().size()));
+
+        return movies;
+    }
+
+    @Override
+    public String rentAMovie(Long movieId, MovieRenter movieRenter) {
+        Movie movieToRent = movieRepository.findById(movieId)
```

Comment:
Ovo mi se ne svidja. Koja greska se vraca ako film ne postoji? 500? Nisam pokrenuo ovo pa ne znam, samo pretpostavljam. Mora da se throw odredjen exception i da se handluje kako treba. Kada budes citao ovo pitaj me da ti objasnim u mom projektu kako je uradjeno

Reply:
https://auth0.com/blog/exception-handling-and-i18n-on-spring-boots-apis/

Imas u mom kodu  ovo uradjeno, ali evo i text o tome